In [92]:
import seaborn as sns; sns.set(style="ticks", color_codes=True)
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression

iris = sns.load_dataset("iris")
all_data = iris[['sepal_length', 'sepal_width', 'petal_length', 'petal_width']]
all_label = iris['species']
train_data, test_data, train_label, test_label = train_test_split(all_data, all_label,test_size=0.5, random_state=1)

model = LogisticRegression(C=1e5, solver='lbfgs', multi_class='ovr')
model.fit(train_data, train_label)
model.score(test_data,test_label)

0.96

In [91]:
model = LogisticRegression(penalty='l2',C=1000, solver='lbfgs', multi_class='ovr')
model.fit(train_data, train_label)
model.score(test_data,test_label)

0.96

|name|liblinear|lbfgs|newton-cg|sag|saga|
|--|--|--|--|--|--|
|OvR|√|√|√|√|√|
|MvM|-|√|√|√|√|
|l1|√|-|-|-|√|
|l2|-|√|√|√|√|

|name|liblinear|lbfgs|newton-cg|sag|saga|
|--|--|--|--|--|--|
|l1&OvR|0.96|-|-|-|0.946|
|l1&MvM|-|-|-|-|0.986|
|l2&OvR|-|0.96|0.96|0.946|0.946|
|l2&MvM|-|0.946|0.946|0.973|0.986|
